src/main/java/eu/luminis/elastic/RestClientFactoryBean.java

```diff
@@ -1,37 +1,29 @@
 package eu.luminis.elastic;
 
-import org.apache.http.Header;
-import org.apache.http.HttpHost;
-import org.apache.http.message.BasicHeader;
+import java.util.Arrays;
+
 import org.elasticsearch.client.RestClient;
-import org.elasticsearch.client.sniff.Sniffer;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.beans.factory.config.AbstractFactoryBean;
 import org.springframework.stereotype.Component;
 
-import java.io.IOException;
+import eu.luminis.elastic.cluster.ClusterManagementService;
 
 /**
  * Factory bean for creating the RestClient instance(s)
  */
 @Component
 public class RestClientFactoryBean extends AbstractFactoryBean<RestClient> {
```

Comment:
Need to think about the @Component here. Due to the classpath scanning for components in the default configuration file, this class will be found as well. Might be better not to include it, we could add a second configuration file that does contain this bean as well as an import of the other configuration file

Reply:
This was done to preserve backwards compatibility with other projects that use this project as a dependency, because they rely on being able to inject the `RestClient`. If this is not a problem then I can refactor some more and use the new `ClusterManagementService` in place of the `RestClient`. Depending projects need to work with the `ClusterManagementService` from then on.